### ISSUE TITLE
Converting rbxm to rbxmx using linux tool corrupts unions

### ISSUE THREAD

**Issue by LabGuy94** [Anaminus/rbxmk#61]
The title says it all, I'm downloading a .rbxm file from roblox and then I'm converting it and when the model gets uploaded again the meshes in the model break.

**Comment by LabGuy94** [comment]
Im using rbxmk run

**Comment by iUnstable0** [comment]
I think you meant 'Converting rbxm to rbxmx', RBXMK is a tool name

**Comment by LabGuy94** [comment]
Yes you are right

**Comment by LabGuy94** [comment]
using rbxmk run

**Comment by Anaminus** [comment]
- What version of rbxmk are you using? You can use the `rbxmk version` command to display the version.
- Can you provide steps that will reproduce the issue? Include a sample .rbxm file, if possible.

**Comment by LabGuy94** [comment]
We are using version 0.8.0 and the first image is what it should look like and the second one is what it looks like at the end. A sample file is also bellow

![image](https://user-images.githubusercontent.com/37680107/156083950-659fce05-d39b-4e30-93ad-006a0d598524.png)

![image](https://user-images.githubusercontent.com/37680107/156083681-133e918e-b1bb-489c-8192-04146abf2102.png)

Bellow is the RBXM file before converted.
https://www.mediafire.com/file/1qnb7fkom7aim7d/HPD.rbxm/file

**Comment by LabGuy94** [comment]
And to reproduce I am just downloading a file, adding a script then reuploading on a separate account

**Comment by LabGuy94** [comment]
And to reproduce I am just converting from rbxm to rbxmx

**Comment by Anaminus** [comment]
One of the problems here is that the MeshIds aren't loading. They aren't loading because, without a descriptor, any property of the Content type can't properly be converted from rbxm to rbxmx. Supplying an up-to-date descriptor will resolve this part of the issue. The easiest way is to include the `--desc-latest` flag, which will download and use the latest API dump from Roblox.

Still unresolved are the wheels, which are made out of unions. I'll look into this part.

**Comment by LabGuy94** [comment]
Ok thanks!

**Comment by iUnstable0** [comment]
I tried with --desc-latest and the mesh still corrupts

**Comment by iUnstable0** [comment]
echo 'the code' | ./rbxmk run - --desc-latest

**Comment by iUnstable0** [comment]
![Screen Shot 2565-03-02 at 11 12 57 PM](https://user-images.githubusercontent.com/46888825/156401599-831fd43a-91f5-4046-8c04-d0c46ec545fb.png)

![Screen Shot 2565-03-02 at 11 14 18 PM](https://user-images.githubusercontent.com/46888825/156401830-08ac3db5-c583-496c-a7bc-da5aa1bb5204.png)

**Comment by LabGuy94** [comment]
I agree with this @iUnstable0

**Comment by Anaminus** [comment]
The flag needs to be added before the file name, or it will be treated as an argument passed to the script. It should look like this:

```bash
echo 'the code' | ./rbxmk run --desc-latest -
```

**Comment by iUnstable0** [comment]
![Screen Shot 2565-03-03 at 9 15 13 AM](https://user-images.githubusercontent.com/46888825/156483260-41bc89d4-55df-49db-8847-99dbf6a714cc.png)
It works now, thanks! but some part are missing I guess it's union

**Comment by LabGuy94** [comment]
Ive renamed this issue to breaks unions because that is now the current bug

**Comment by Anaminus** [comment]
It turns out that the issue with unions is the same as with MeshParts. When a union is constructed, an asset is uploaded representing the mesh of the union. In this case, the ID is stored in the hidden "AssetId" property. Once again, the property is incorrectly converted to a string instead of the Content type, resulting in the mesh failing to load.

The problem is that this property is not included in Roblox's API dump. Fortunately, I've anticipated this sort of thing happening, with a system that allows descriptors to be patched with additional data. 

Because this will be a common problem, I've produced a patch file that includes missing properties, along with instructions on how to apply it:

https://github.com/Anaminus/rbxmk/tree/imperative/doc/examples/dump-patch

I've confirmed on my end that this fixes the unions failing to load.
